### DISCUSSION TITLE
Is RST pin necessary in 8-bit parallel mode?

### DISCUSSION THREAD

**Discussion by sandric** [Bodmer/TFT_eSPI#2442] [Q&A - General] [answered]
Hi, mb dumb question, but while trying to shrink my little project to work on tiny RP2040 Zero board that has only 20 pins (if not soldered as smd which I can not do at the moment) - I tried to unplug RST pin, and it looks like my 7 inch SSD1963 tft display, working in 8-bit mode, works just fine, I can draw different pictures of SD card pressing my keypad..

Can someone clarify why is it possible, and what exactly it does - I guess I'll be limited in doing smth not having it connected to my MCU?

Below is my `UserSetup.h` file, if will be needed by any chance:

```C

#define SSD1963_800ALT_DRIVER
#define TFT_RGB_ORDER TFT_BGR

#define TFT_DC   8
#define TFT_WR   9
#define TFT_RST  10 // <-- RST pin is declared here, but not used actually

#define TFT_D0   0
#define TFT_D1   1
#define TFT_D2   2
#define TFT_D3   3
#define TFT_D4   4
#define TFT_D5   5
#define TFT_D6   6
#define TFT_D7   7

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT
```

**Comment by Bodmer** [comment, marked as answer]
The display has 2 reset methods.
1. Via the hardware RESET line
2. Via a software command.
The library uses both methods. In some cases garbage on the bus while the processor powers up and the logic lines and voltage rails get established causes the software reset method to fail, in this case a hardware reset is needed after power up to get the display in a sensible state for communication.

If you do not connect the reset line to a GPIO pin then pull it up or connect to 3.3V.

Other ways to save pins: 
1. With the RP2040 the display cannot be read via the parallel bus, so the RD line must be pulled to 3.3V
2. The chip select line (CS) can be tied low to 0V and not connected to a GPIO. However this can lead to garbage on the bus upsetting the display at power up. So in this case the reset line should be driven with a GPIO.
3. Some processors have a Power On Reset (POR) output that resets the p[rocessor at power up, this can be used to also connect to the display reset.

**Reply by sandric** [reply]
Thanks a lot for explanation, I will pull RST to 3.3v! Yeah, I already not using RD line, CS too since I'm using parallel mode, and it also appears I even don't have in on my pins for controller. Although I'm using CS pin for SD card SPI interface - the only SPI bus on my TFT+TouchScreen+SD Card "all in one module" actually, but trying to ignore it seems not to work and SD card not responses, I'll try to pull it to ground though.

**Reply by sandric** [reply]
Ok, I'm dumb, I pulled CS pin of SD card to ground and not just floating - and it seems to work, thanks again a lot, now I have even 1 pin to spare on my tiny rp zero xD
